paper/paper.md

```diff
@@ -36,16 +36,28 @@ bibliography: paper.bib
 
-# Dataconverter and validation (Sherjeel)
+# Dataconverter and validation
 
-      * Mechanism to write an own reader, i.e. pynxtools-plugin mechanism and test frameworks, not mentioned to much about pynxtools-plugins
-      * ELN Generator - one sentence or so
+The __dataconverter__ module forms the core of pynxtools, combining instrument output files and data from electronic lab notebooks into NeXus-compliant HDF5 files. The converter performs three key operations: reading experimental data through specialized readers, validating against NeXus application definitions to ensure compliance with existence, shape, and format constraints, and writing valid NeXus/HDF5 output files.
+
+The __dataconverter__ provides a CLI interface to produce NeXus files where users can use one of the built-in readers for generic functionality or technique-specific reader plugins distributed as separate Python packages.
+
+For developers, the __dataconverter__ provides an abstract __reader__ class to create plugins that process experiment-specific file formats and try to fill the NeXus specification. A __Template__ object is passed to the __reader__ by the __dataconverter__ that acts like a form that has to be filled by the __reader__. __Template__ is a subclass of a regular Python __dict__ class. It keeps a similar interface to a __dict__ to help developers write code they are familiar with. It transparently ensures structural compliance with the selected NeXus application definition. __Template__ categorizes all data elements according to three requirement levels: required, recommended, and optional.
+
+This allows the __dataconverter__ to control what the reader plugins provide and validate this against the selected NeXus application definition. Once, the reader plugin returns a __Template__ object, the __dataconverter__'s validation routine checks if all required fields in the NeXus application definition exist. This includes complex dependency relationships such as inheritance chains or required children within optional parent groups. The validation then collects all data entities that have a specification and ensures data integritiy by verifying the data type, shape, and constraints set by the specification. The __dataconverter__'s validation routine throws an error if there is a required data entity it found invalid. It also emits warnings to the CLI for the user and/or developer for any data it invalidates or doesn't find a specification for. This helps the users identify issues while practically building a NeXus file.
```

Comment:
"integritiy" > "integrity"

Reply:
"it doesn't find a" that is not defined in the application definition

Reply:
"specification" here refers to a single data point. It could be the specification for a field, group, attribute, unittype, requirements. It's a more general word here.